napari/layers/labels/labels.py

```diff
@@ -241,6 +241,25 @@ def num_colors(self, num_colors):
         self._selected_color = self.get_color(self.selected_label)
         self.events.selected_label()
 
+    def _get_state(self):
+        """Get dictionary of layer state.
+
+        Returns
+        -------
+        state : dict
+            Dictionary of layer state.
+        """
+        state = self._get_base_state()
+        state.update(
+            {
+                'is_pyramid': self.is_pyramid,
+                'num_colors': self.num_colors,
+                'seed': self.seed,
+                'data': self.data,
+            }
+        )
```

Comment:
These dictionaries feel quite redundant and brittle. I would suggest that we have some kind of global/class variable, e.g. `_SERIALIZED_ATTRS`, and then this becomes:

```python
state.update({name: getattr(self, name) for name in self._SERIALIZED_ATTRS})
```

Reply:
unfortunately, there's at least [one case with colormaps](https://github.com/sofroniewn/napari/blob/65a275a168ae55fef38e4568bbf6178d7822d724/napari/layers/image/image.py#L353) where this won't work because it's not just getting the attribute.  could just change the colormap attribute though

Reply:
ooh nice, I like that idea. i feel like maybe we should change the colormap attribute for that reason, I've always found the tuple a little unclear. maybe I'll make a short follow-up PR

Reply:
sounds good. Well, the colormap itself should be saved anyway, since the name doesn't always contain all the info. Do vispy colormaps serialize, though?

Reply:
I don't know if vispy colormaps serialize - we might want to move to some numpy / list of numpy like representation if we can - like in #782 - then our model files would be vispy independent which I think would be really nice